File: mgyoutube-multitech/api-dotnet/modules/SearchesModule.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace api_dotnet.modules
{
    public interface SearchesModule
    {
        Task<List<string>> GetSavedSearchesForParent(string parentUsername);

        Task AddSearchToParent(string parentUsername, string searchPhrase);

    }
}

File: mgyoutube-multitech/api-java/src/main/java/com/djpedesen/mgyoutube/api_java/modules/SearchesModule.java
package com.djpedesen.mgyoutube.api_java.modules;

import java.util.List;

public interface SearchesModule {

	List<String> getSavedSearchesForParent(String parentUsername) throws Exception;

	void addSearchToParent(String parentUsername, String searchPhrase) throws Exception;

}
